packages/relay/src/formatters.ts

```diff
@@ -232,6 +237,36 @@ const stringToHex = (str) => {
   return hex;
 };
 
+const formatHex = (hexValue: string): string => (hexValue === '0x' ? '0x0' : hexValue);
+
+const convertToHex = (value: number | string | null): string => {
+  // Check for null value
+  if (value === null || value === undefined) {
+    return '0x0';
+  }
+
+  // Check and convert decimal number to hex
+  if (typeof value === 'number') {
+    return '0x' + value.toString(16);
+  }
+
+  // Check if the value is a hex string
+  if (typeof value === 'string') {
+    // Validate hex string
+    if (/^0x[0-9a-fA-F]+$/.test(value)) {
+      return value;
+    }
+    // If it's a string of a decimal number, convert it
+    if (/^\d+$/.test(value)) {
+      return '0x' + parseInt(value, 10).toString(16);
+    }
+    throw new Error("Invalid input string format. Must be a decimal string or hex string with '0x' prefix.");
+  }
+
+  // Throw error if input type is unexpected
+  throw new Error('Invalid input type. Expected number, string, or null.');
```

Comment:
I think this is not necessary. The `value` argument itself only accept `number | string | null`, and your `if statements` already cover all those three scenarios, so this error will never be reached.

Reply:
Removed.